Source/WebCore/platform/graphics/cg/GradientRendererCG.cpp

```diff
@@ -579,11 +579,18 @@ void GradientRendererCG::Shading::shadingFunction(void* info, const CGFloat* raw
     float offset = (stop1.offset == stop0.offset) ? 0.0f : (requestedOffset - stop0.offset) / (stop1.offset - stop0.offset);
 
     // 3. Interpolate the two stops' colors by the computed offset.
-    // FIXME: We don't want to due hue fixup for each call, so we should figure out how to precompute that.
-    auto interpolatedColor = interpolateColorComponents<alphaPremultiplication>(
-        std::get<InterpolationSpace>(data->colorInterpolationMethod().colorSpace),
-        makeFromComponents<InterpolationSpaceColorType>(stop0.colorComponents), 1.0f - offset,
-        makeFromComponents<InterpolationSpaceColorType>(stop1.colorComponents), offset);
+    auto interpolatedColor = [&]() {
+        if (stop0.offset == 0.0f && data->firstStopIsSynthetic())
+            return makeFromComponents<InterpolationSpaceColorType>(stop0.colorComponents);
+
+        if (stop1.offset == 1.0f && data->lastStopIsSynthetic())
+            return makeFromComponents<InterpolationSpaceColorType>(stop1.colorComponents);
```

Comment:
I think this needs a comment. To me, these things are not obvious:

1. why this is needed and doesn’t do the right thing without a special case
2. why it’s important to do this only for _synthetic_ stops
3. why this shouldn’t be done in `interpolateColorComponents` instead of here, maybe the fact that the _synthetic_ state isn’t passed through, maybe the fact that this is needed only for CG
4. why having this special case at exactly 0 and 1 doesn’t create discontinuities in the gradient

I think we should consider saying something brief in the code that might clarify rather than relying solely on test cases and change log.